src/python/pants/backend/experimental/jvm/util_rules.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import hashlib
+from dataclasses import dataclass
+
+from pants.engine.fs import Digest, DigestContents, DigestSubset, FileDigest, PathGlobs
+from pants.engine.rules import Get, collect_rules, rule
+
+
+@dataclass(frozen=True)
+class ExtractFileDigest:
+    digest: Digest
+    file_path: str
+
+
+@rule
+async def digest_to_file_digest(request: ExtractFileDigest) -> FileDigest:
```

Comment:
Maybe add TODO comment here that this should become an engine intrinsic?

Reply:
Done